src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -125,15 +125,15 @@ public async Task Invoke(HttpContext context)
         }
 
         // Assumes the response headers have not been sent.  If they have, still attempt to write to the body.
-        private Task DisplayException(HttpContext context, Exception ex)
+        private Task DisplayException(ErrorContext errorContext)
         {
-            var compilationException = ex as ICompilationException;
+            var compilationException = errorContext.Exception as ICompilationException;
```

Comment:
Factor out DisplayCompilationException as a filter? Even if you don't register it via DI.

Reply:
Why?

Reply:
Proof of concept. This is doing exactly what filters are designed to do.

Reply:
I don't think it makes any sense. The thing left to resolve is the ordering concern that @rynowak brought up. The slight code make the diff bigger and don't add any value.